src/platform/xcode_templates.nit

```diff
@@ -0,0 +1,681 @@
+#
+# This implementation simply counts upward from 0.
+class PbxUUIDGenerator
+	private var seed = 0
+
+	# Generate a new UUID
+	fun next_uuid: String
+	do
+		seed += 1
+
+		var hex_val = seed.to_hex.to_upper
+		return "0"*(24-hex_val.length) + hex_val
+	end
+end
+
+# Reference to a file for the PBX format of a project file
+#
+# TODO create subclasses for different file types, this is currently for
+# compilable source files only.
+class PbxFile
+
+	# Path to `self`
+	var path: String
+
+	# UUID for build elements
+	private var build_uuid: String = sys.pbx_uuid_generator.next_uuid is lazy
+
+	# File reference UUID
+	private var ref_uuid: String = sys.pbx_uuid_generator.next_uuid is lazy
+
+	# Documentation to add besides this file in the template
+	private fun doc: String do return path
+
+	# PBX file type for `self`
+	fun file_type: String
+	do
+		var map = sys.pbx_file_types
+		var ext = path.file_extension
+		if ext != null and map.keys.has(ext) then return map[ext]
+		return "unknown"
+	end
+
+	# PBX description of this file
+	private fun description: Writable do return """
+		{{{ref_uuid}}} /* {{{doc}}} */ = {
+			isa = PBXFileReference;
+			fileEncoding = 4;
+			lastKnownFileType = {{{file_type}}};
+			path = {{{path}}};
+			sourceTree = "<group>";
+			};
+"""
+
+	private fun add_to_project(project: PbxprojectTemplate)
+	do
+		project.source_files.add self
+		project.files.add self
+	end
+end
+
+# Template for a PBX project file, usually a `project.pbcproj`
+#
+# This file list all information required to build an XCode project.
+# It would usually be written and read by XCode.
+# From the command line, xcodebuild can read this file but not write it.
+#
+# Information in the file (simplified list):
+#
+# * Compilable source files
+# * Asset files
+# * Build configurations (Release and debug modes, cflags, etc.)
+# * List of files composing the project
+class PbxprojectTemplate
+	super Template
+
+	# Name of the project
+	var name: String
+
+	# All body/implementation source files to be compiled
+	private var source_files = new Array[PbxFile]
+
+	# All asset files added to the app package
+	private var asset_files = new Array[PbxFile]
+
+	# All files used by this project
+	private var files = new Array[PbxFile]
+
+	# Add `file` to this project
+	fun add_file(file: PbxFile) do file.add_to_project(self)
+
+	redef fun rendering
+	do
+		add """
+// !$*UTF8*$!
+{
+	archiveVersion = 1;
+	classes = {
+	};
+	objectVersion = 46;
+	objects = {
+
+/* Begin PBXBuildFile section */
+"""
+
+		# List build files (compilable sources and assets) with their reference UUID
+		for array in [source_files, asset_files] do for file in array do add """
+		{{{file.build_uuid}}} /* {{{file.doc}}} */ = {
+			isa = PBXBuildFile;
+			fileRef = {{{file.ref_uuid}}} /* {{{file.doc}}} */;
+			};
+"""
+
+		add """
+/* End PBXBuildFile section */
+
+/* Begin PBXContainerItemProxy section */
+		AF9F83EA1A5F0D21004B62C0 /* PBXContainerItemProxy */ = {
```

Comment:
XCode generate them using the date, a local computer id, and stuff like that. But in practice it only needs to be unique within the same project.

In this implementation we do not have collision with `build_uuid` as they use the same generator. The static UUIDs in the file all begin with AF (It was 0F but I had to change it because of #1223), they are high enough that we will never have enough files to collide.

The only possible problem is when using a library and thus having 2 projects, I don't know if the UUIDs may collide then. I'll keep an eye out for that.